src/main/groovy/de/dkfz/roddy/execution/io/ExecutionResult.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.execution.io;
+
+import de.dkfz.roddy.core.InfoObject;
+
+import java.util.List;
+
+/**
+ * Stores the result of a command execution.
+ * Commands can i.e. be executed via ssh or on the local command line.
+ * @author michael
+ */
+public class ExecutionResult extends InfoObject {
+
+    /**
+     * Successful or not?
+     */
+    public final boolean successful;
+    /**
+     * An error number. Normally the return code for shell stuff.
+     */
+    public final int errorNumber;
+    /**
+     * All result lines.
+     */
+    public final List<String> resultLines;
+    /**
+     * First line of the result array.
+     * Null if no entries are in the array.
+     */
+    public final String firstLine;
```

Comment:
In this case, I doubt the optimization will actually yield an improvement -- neither for the emptyness check nor for getting the first line is an expensive operation (O(1)).